### DISCUSSION TITLE
How to add customlink while dragging from one node to another..

### DISCUSSION THREAD

**Discussion by snehabihani** [Blazor-Diagrams/Blazor.Diagrams#201] [Q&A]
How to add customlink while dragging from one node to another..
I have created customlink model. With this Diagram.Links.Add(new NodeLink(node.GetPort(PortAlignment.Bottom), node1.GetPort(PortAlignment.Top))); I can draw On page load but i want on drag functionality.

**Comment by zHaytam** [comment]
Hello, I assume `NodeLink` is your class that inherits from `BaseLinkModel` right?
If yes, then you can set a factory in the options in order to create the links and control how:
```
var options = new DiagramOptions
{
    Links = new DiagramLinkOptions
    {
        Factory = (diagram, sourcePort) => new NodeLink(sourcePort, null) // Not attached yet
    } 
}
```

**Comment by snehabihani** [comment]
Hi, thank you for quick reply. But I want to change the link color for
every node.. means for different nodes there will be different link color..

On Friday, May 27, 2022, Haytam Zanid ***@***.***> wrote:

> Hello, I assume NodeLink is your class that inherits from BaseLinkModel
> right?
> If yes, then you can set a factory in the options in order to create the
> links and control how:
>
> var options = new DiagramOptions
> {
>     Links = new DiagramLinkOptions
>     {
>         Factory = (diagram, sourcePort) => new NodeLink(sourcePort, null) // Not attached yet
>     }
> }
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Blazor-Diagrams/Blazor.Diagrams/discussions/201#discussioncomment-2834703>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM4MS63MKKDS4XQWCVVHE2DVMDKLRANCNFSM5XD6KMMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> <Blazor-Diagrams/Blazor.Diagrams/repo-discussions/201/comments/2834703@
> github.com>
>


-- 
Regards,
Sneha Bihani
Mob:9960298364

**Comment by zHaytam** [comment]
Based on the source node?

**Reply by snehabihani** [reply]
Depend on port.. In my Application there are multiple ports for every node.. So depending on the port I need different colors on drag

**Reply by snehabihani** [reply]
Hi I got the answer.. But now I am creating custom link 
 **public class NodeLink : LinkModel
    {
        public NodeLink(PortModel sourcePort, PortModel targetPort = null) : base(sourcePort, targetPort)
        {
            TargetMarker = LinkMarker.Circle;
        }
    }**
    
    And Assigning like

     **Diagram.Options.Links.Factory = (diagram, port) => new NodeLink(port);
        Diagram.Links.Added += (l) =>
        {
            if (l is NodeLink nodeLink)
            {
                if (nodeLink.SourcePort != null)
                {
                    nodeLink.Color = ((ColoredPort)nodeLink.SourcePort).ExecutionPortdefinition.LinkColor;
                }
            }
            l.TargetMarker = LinkMarker.Arrow;
        };**
        
        But target marker is not visible..
